yspryn/hw9/fraction.py

```diff
@@ -0,0 +1,132 @@
+from math import gcd
+
+
+class Fraction:
+
+    def __init__(self, num: int, denom: int = 1):
+
+        if denom == 0:
+            raise ZeroDivisionError
+
+        _gcd = gcd(num, denom)
+        self.num = num // _gcd
+        self.denom = denom // _gcd
+
+        if self.denom < 0:
+            self.num *= -1
+            self.denom *= -1
+
+    @classmethod
+    def convert(cls, item):
+        if isinstance(item, int):
+            return cls(item, 1)
+        elif isinstance(item, cls):
+            return item
+
+    def __repr__(self):
+        return f"Fraction({self.num}, {self.denom})"
+
+    def __str__(self):
+        return f"\'{self.num} / {self.denom}\'"
+
+    # division
+    def __truediv__(self, other):
+        value1 = Fraction.convert(self)
+        value2 = Fraction.convert(other)
+        if (value1.num * value2.denom) % (value1.denom * value2.num) == 0:
+            return (value1.num * value2.denom) / (value1.denom * value2.num)
+        else:
+            return Fraction(value1.num*value2.denom, value1.denom*value2.num)
```

Comment:
39 line is correct, 
line 37 is redundant, 

since it is operation of fraction you want to return fraction object as a result of operation